src/game.cpp

```diff
@@ -2281,6 +2281,88 @@ int game::get_temperature()
     return temperature + m.temperature(u.posx, u.posy);
 }
 
+int game::get_radiant_temperature(int posx, int posy)
+{
+    int felt_radiant_temperature = 0;    
+    
+    /**
+    *   Sun energy (distance is constant due to its magnitude)
+    **/
+    
+    if (weather == WEATHER_SUNNY && is_in_sunlight(posx, posy)) {
+        felt_radiant_temperature += 1000;
+    }
+    else if (weather == WEATHER_CLEAR && is_in_sunlight(posx, posy)) {
+        felt_radiant_temperature += 500;
+    }
+    
+    int tile_distance = 1;
+    int t;
+    
+    int fire_temperature_level[4] = {1000, 2000, 4000, 8000};
+    int lava_temperature = fire_temperature_level[3];
+    
+    for (int j = -6 ; j <= 6 ; j++) {
+        for (int k = -6 ; k <= 6 ; k++) {
+        
+            // Skip things that can't be seen
+            if (!m.sees(posx, posy, posx + j, posy + k, -1, t)) {
+                continue;
+            }
+        
+            auto &tile_field = m.field_at(posx + j, posy + k);
+            tile_distance = std::max(1, std::max( std::abs( j ), std::abs( k ) ) );
+            int tile_radiant_temperature = 0;
+        
+            /**
+            *   Fire energy
+            **/                  
+        
+            if (tile_field.findField(fd_fire)) {
+                switch (tile_field.findField(fd_fire)->getFieldDensity()) {
+                    case 1:
+                        tile_radiant_temperature += fire_temperature_level[0]; break;
+                    case 2:
+                        tile_radiant_temperature += fire_temperature_level[0]; break;
+                    case 3:
+                        tile_radiant_temperature += fire_temperature_level[0]; break;
+                    case 4:
```

Comment:
Field density can only go to 3. Is it intended that every case block is exactly the same (same index in the `fire_temperature_level` array)?